### PR TITLE
adding react-onsenui typings

### PR DESCRIPTION
**Pull request by gldfdp**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: react-onsenui/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,377 @@
+// Type definitions for React OnSenui (react-onsenui) 2.1
+// Project: https://onsen.io/v2/docs/guide/react/
+// Definitions by: Ozytis <https://ozytis.fr>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+import {
+    Component
+} from 'react';
+
```
Comment (yuit) [inline react-onsenui/index.d.ts]: please remove these extra lines

File: react-onsenui/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+
+
+
+    declare interface Modifiers_string {
```
Comment (yuit) [inline react-onsenui/index.d.ts]: Could you remove these indentations

File: react-onsenui/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{	
+	"compilerOptions": {
```
Comment (yuit) [inline react-onsenui/tsconfig.json]: is these indentation a 4 spaces

File: react-onsenui/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+        swipeTargetWidth?: number,
+        width?: number,
+        animation?: "overlay" | "default"
+        animationOptions?: AnimationOptions,
+        openThreshold?: number,
+        mode?: "collapse" | "split"
+    }, any>{
+
+    }
+
```
Comment (yuit) [inline react-onsenui/index.d.ts]: Should this be just one single line

File: react-onsenui/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+        isSwipeable?: boolean,
+        swipeTargetWidth?: number,
+        width?: number,
+        animation?: "overlay" | "default"
+        animationOptions?: AnimationOptions,
+        openThreshold?: number,
+        mode?: "collapse" | "split"
+    }, any>{
+
+    }
```
Comment (yuit) [inline react-onsenui/index.d.ts]: Just put "}" of this empty class in the same line as "{"

**Comment by yuit** [comment]
@gldfdp thanks for the PR. just have couple comments on style but otherwise look good

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Pull request merged**